### PR TITLE
[Babel 8] Remove `enums` option of flow plugin

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | https://github.com/babel/babel/issues/16456 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | https://github.com/babel/website/pull/2965
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
We can also do this in the next minor

**Comment by liuxingbaoyu** [comment]
Isn't this a breaking change?

**Comment by nicolo-ribaudo** [comment]
In which case can it break some existing usage?

File: packages/babel-plugin-syntax-flow/src/index.ts [hunk 1]
```diff
@@ -2,22 +2,30 @@ import { declare } from "@babel/helper-plugin-utils";
...
+    if (enums !== undefined) {
+      throw new Error(
+        "The .enums option has been removed and it's now always enabled. Please remove it from your config.",
+      );
+    }
+  } else {
+    if (typeof enums !== "boolean" && typeof enums !== "undefined") {
+      throw new Error(".enums must be a boolean, or undefined");
+    }
   }
```
Comment (liuxingbaoyu) [inline packages/babel-plugin-syntax-flow/src/index.ts:29]: Now the `enums` option will have no effect, and we also explicitly throw an error here to align with ts's `allowDeclareFields`.

**Review by nicolo-ribaudo — APPROVED** [review]
This looks good, but in the next minor we should enable the option by default.

File: packages/babel-preset-typescript/test/fixtures/flow-compat/js-invalid/input.js
Comment (liuxingbaoyu) [inline packages/babel-preset-typescript/test/fixtures/flow-compat/js-invalid/input.js:1]: I wasn't sure the purpose of this test so just removed it.

Comment (nicolo-ribaudo) [reply packages/babel-preset-typescript/test/fixtures/flow-compat/js-invalid/input.js:1]: I think it was to make sure that  we can parse valid non-TS syntax that would be invalid in TS files.

**Pull request merged**